samples/java-eventsourced-counter-pojo/src/main/java/org/example/domain/CounterState.java

```diff
@@ -0,0 +1,29 @@
+package org.example.domain;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class CounterState {
+
+  public final int value;
+
+  @JsonCreator
+  public CounterState(@JsonProperty("value") int value) {
+    this.value = value;
+  }
+  
+  public static CounterState defaultInstance() {
+    return new CounterState(0);
+  }
```

Comment:
This is just mimicing the generated proto Javas, maybe just instantiate with 0 in the entity instead of this?

And also, if we really want to appeal to Java devs, perhaps the class should be mutable? I _think_ that should be safe since there is a single writer (but maybe we do future juggling in the  component impls so it isnt)

Reply:
For the moment the samples are intended for us only. It will take some time before we consider that the POJO style is the one we want to promote the most. We can review this later. 

I personally think it's ok to promote immutable classes. I've seen immutable classes showing up in Java blog posts. Not saying that this is the most common idiom in Java world, but it's not as exotic as it used to be.

Reply:
removed the `defaultinstance()` and using default constructor now.